File: app/src/main/java/com/example/islamic/view/Azkar.java
package com.example.islamic.view;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.example.islamic.R;
import com.example.islamic.databinding.ActivityAzkarBinding;

public class Azkar extends AppCompatActivity {

    private ActivityAzkarBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityAzkarBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        binding.cardAzkarAlsabah.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(Azkar.this, Azkar_Sabah.class);
                startActivity(intent);
            }
        });

        binding.cardAzkarAlmassa.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(Azkar.this, Azkar_Massa.class);
                startActivity(intent);
            }
        });

        binding.cardAzkarAfter.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(Azkar.this, Azkar_After.class);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/eslamdev/islamic/presentation/ui/activity/Azkar.kt
package com.eslamdev.islamic.presentation.ui.activity

import android.content.Intent
import android.os.Bundle
import android.widget.ImageButton
import androidx.appcompat.app.AppCompatActivity
import com.eslamdev.islamic.R
import com.eslamdev.islamic.databinding.ActivityAzkarBinding
import com.eslamdev.islamic.domain.repository.AzkarType

class Azkar : AppCompatActivity() {

    private lateinit var binding: ActivityAzkarBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityAzkarBinding.inflate(layoutInflater)
        setContentView(binding.root)


        binding.btnBack.setOnClickListener {
            finish()
        }

        binding.cardAzkarAlsabah.setOnClickListener {
            openAzkarDetails(AzkarType.SABAH)
        }

        binding.cardAzkarAlmassa.setOnClickListener {
            openAzkarDetails(AzkarType.MASSA)
        }

        binding.cardAzkarAfter.setOnClickListener {
            openAzkarDetails(AzkarType.AFTER_PRAYER)
        }
    }

    private fun openAzkarDetails(type: AzkarType) {
        val intent = Intent(this, AzkarDetailsActivity::class.java)
        intent.putExtra("EXTRA_AZKAR_TYPE", type.name)
        startActivity(intent)
    }
}